File: MatchVision/backend/models/touch.py
from enums.fundamentals import FundamentalType
from enums.touch_results import TouchResult

# one of the players touches the ball
class Touch:
    def __init__(self, player: int, fundamental: FundamentalType, outcome: TouchResult):
        self.player = player
        self.fundamental = fundamental
        self.outcome = outcome

File: MatchVision/src/app/Models/Touch.ts
export enum FundamentalType {
    SERVE = 'Serve',
    SERVE_RECEIVE = 'Serve_Receive',
    SET = 'Set',
    SPIKE = 'Spike',
    BLOCK = 'Block',
    DEFENSE = 'Defense'
}

export enum TouchResult {
    POSITIVA = 'Positiva',
    BUONA = 'Buona',
    NEUTRA = 'Neutra',
    NEGATIVA = 'Negativa',
    ERRORE = 'Errore',
}

export class Touch {
    constructor(
        public id: number,
        public set: number,
        public fundamental: FundamentalType | string,
        public outcome: TouchResult | string,
        public player: number
    ) {}
}